src/utilities/debouce.ts

```
@@ -0,0 +1,26 @@
+ *
+ * Useful for optimizing performance when dealing with events like
+ * window resizing, input typing, or API calls on user interaction.
+ *
+ * @param fn
+ * @param delay
+ *
+ */
+
+export function debounce<T extends (...args: string[]) => void>(
+    fn: T,
+    delay: number = 300,
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)